cores/esp32/esp32-hal-uart.c

```diff
@@ -341,7 +341,9 @@ uint32_t uartAvailable(uart_t* uart)
         return 0;
     }
 #ifdef UART_READ_RX_FIFO
+    UART_MUTEX_LOCK();
     return (uxQueueMessagesWaiting(uart->queue) + uart->dev->status.rxfifo_cnt) ;
+    UART_MUTEX_UNLOCK();
```

Comment:
this line will never be called and the lock will potentially be left locked. You will need to store the value in a temporary variable and unlock prior to return.

Reply:
Absolutelly right!